### PR TITLE
Rename GlobalContextLite

### PR DESCRIPTION
**Pull request by gabriel**
I see you made an interface that defines the context for proofs, but you called it `GlobalContextLite`.

If it's the context thats needed for proofs, then we should call it ProofContext. This is important because other things shouldn't use it if they aren't proof contexts. Others might see GlobalContextLite and add to it and then over time you'll have another shared "global" interface that is just a smaller kitchen sink then the one we had before.

I did this for other packages, like `install`:
https://github.com/keybase/client/blob/master/go/install/install.go#L32-L42
so it doesn't have any dependency on GlobalContext either.

### CODE REVIEW

**Comment by gabriel** [comment]
cc @maxtaco @patrickxb

**Comment by maxtaco** [comment]
Ok! We considered this name among others but didn't have a strong opinion.  Happy to go with this name

:+1:

**Comment by patrickxb** [comment]
My argument against calling it ProofContext is that there is nothing about it specific to proofs, and there is no reason other things couldn't use it.

**Comment by maxtaco** [comment]
I'm happy to flip a coin if you guys want!

**Comment by gabriel** [comment]
I guess my argument is:

(a) I think it does have something to do with proofs in that it defines the environment or context  needed to do proof stuff?

(b) And others shouldn't use it (unless they are in the contexts of proofs). If they have another set of methods they need from the "global" context (app), they should define the things they need in a new interface. Otherwise slowly over time GlobalContextLite isn't so lite as more and more people add to it.

And the reason I think it's important is we have issues with dependencies (both internal and external) and I think using interfaces more like this (instead of sharing "global" interfaces) helps us untangle that a bit. (Defining interfaces that list only the things they need. This might be a pipe dream, but I think worth trying.)

**Comment by maxtaco** [comment]
Maybe what @patrickxb is saying is that if we have ~5 libraries that take parts of the old `libkb`, then each is going to have something like the `GlobalContextLite`/`ProofContext` and they're going to be all the same...

**Comment by patrickxb** [comment]
Yes, that was my reason for something like BaseContext.  

It could then be embedded in other contexts that need more functions, not have further refactors add methods to BaseContext.

It just has logging and API.  Something down the road that wants logging and API will see ProofContext and either use it, which would be a weird name, or not use it and make a new one with logging and API.

**Comment by gabriel** [comment]
Yeah I think someone would create a new interface in that case (and that would be good)... even if they were using the same "features" of the app because, in the future they may diverge. Having duplicate or similar interfaces is way better than a bad abstraction?

Like when you make a base context you are saying, everything in the future will use these features/methods. So for example, proofs need GetAPI(). But not everything well need to use that. I think maybe the only thing in a BaseContext that everything would probably use is GetLog and that's probably it?

To me this is ok:

```go
type ProofContext interface {
  GetLog()
  GetAPI()
}

type InstallContext interface {
  GetLog()
  GetFusePath()
}

type ServiceContext interface {
  GetLog()
  GetAPI()
  GetSocketOrSomething()
}
```

We could do something like:

```go
type BaseContext interface {
  GetLog() ...
}

// APIContext defines methods for accessing API. It makes sense that any API implementation would need to expose these methods.
type APIContext interface {
  BaseContext
  GetAPI() ...
  GetExternalAPI() ...
  GetServerURI() ...
}

type ProofContext interface {
  APIContext
}

type InstallContext interface {
  BaseContext
  GetFusePath()
}
```

tl;dr Code duplication over bad abstractions, composition over inheritance, be precise with interfaces

**Comment by patrickxb** [comment]
I like your second plan.  

Could rename BaseContext to LogContext, and remove embedding Base/LogContext from APIContext.  Then ProofContext would be

    type ProofContext interface {
        LogContext
        APIContext
    }

**Comment by maxtaco** [comment]
I'm cool with this too!

**Comment by gabriel** [comment]
Yeah I like this. I think this is thumbable now?

**Comment by maxtaco** [comment]
👍 (but there's a CI failure)

**Comment by patrickxb** [comment]
LGTM.  thanks for making this better @gabriel!

**Comment by gabriel** [comment]
The circle CI is a known failure being worked on.
